src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7970,89 +7970,121 @@ s32 Player_ActionHandler_13(Player* this, PlayState* play) {
+                // At this point the talk offer can be accepted.
+                // "Speak" or "Check" will appear on the A button in the HUD.
+                if ((lockOnActor == NULL) || (lockOnActor == talkOfferActor)) {
+                    this->stateFlags2 |= PLAYER_STATE2_CAN_ACCEPT_TALK_OFFER;
+                }
 
-                            if (var_t1 || !var_t2) {
-                                var_a1->textId = ABS_ALT(this->tatlTextId);
-                            } else if (var_a1->hintId != 0xFF) {
-                                var_a1->textId = var_a1->hintId + 0x1900;
-                            }
-                        }
+                if (!CutsceneManager_IsNext(CS_ID_GLOBAL_TALK)) {
+                    return false;
+                }
 
-                        this->currentMask = D_80862B2C;
-                        gSaveContext.save.equippedMask = this->currentMask;
-                        Player_TalkWithPlayer(play, talkActor);
-                        return true;
+                if (CHECK_BTN_ALL(sPlayerControlInput->press.button, BTN_A) ||
+                    (talkOfferActor->flags & ACTOR_FLAG_TALK_OFFER_AUTO_ACCEPTED)) {
+                    // Talk Offer has been accepted.
+                    // Clearing `cUpTalkActor` guarantees that `talkOfferActor` is the actor that will be spoken to
+                    cUpTalkActor = NULL;
+                } else if (cUpTalkActor == NULL) {
+                    return false;
+                }
+            }
+
+            if (cUpTalkActor != NULL) {
+                if (!forceTalkToTatl) {
+                    this->stateFlags2 |= PLAYER_STATE2_200000;
+                    if (!CutsceneManager_IsNext(CS_ID_GLOBAL_TALK) ||
+                        !CHECK_BTN_ALL(sPlayerControlInput->press.button, BTN_CUP)) {
+                        return false;
                     }
                 }
+
+                talkOfferActor = cUpTalkActor;
+                this->talkActor = NULL;
+
+                if (forceTalkToTatl || !canTalkToLockOnWithCUp) {
+                    cUpTalkActor->textId = ABS_ALT(this->tatlTextId);
+                } else if (cUpTalkActor->hintId != 0xFF) {
+                    cUpTalkActor->textId = cUpTalkActor->hintId + 0x1900;
+                }
             }
+
+            // `sSavedCurrentMask` saves the current mask just before the current action runs on this frame.
+            // This saved mask value is then restored just before starting a conversation.
+            //
+            // This handles an edge case where a conversation is started on the same frame that a mask was taken on or
+            // off. Because Player updates early before most actors, the text ID being offered comes from the previous
+            // frame. If a mask was taken on or off the same frame this function runs, the wrong text will be used. This
+            // is especially important to prevent unwanted behavior with regards to mask trading.
```

Comment:
I dont think mask trading is in this game, at least not in the same way it is in oot. Might be worth trying to change it to be more generic, or just remove the last sentence

Reply:
Oh, nice catch. Thanks

Reply:
In OoT I mention this because not only is the wrong text is used, but also the action that comes after will be wrong. Like for example, the guard in kakariko softlocks when trying to open the gate cause his actions that come after the text are a bit thrown off.

I dont know off the top of my head if there are any text -> action flows that will be messed up by this glitch, but something to think about